charm-slurmd/src/charm.py

```diff
@@ -183,6 +195,37 @@ def _on_slurmctld_available(self, event):
         self._on_set_partition_info_on_app_relation_data(event)
         self._check_status()
 
+        # check etcd for hostnames
+        self.on.check_etcd.emit()
+
+    def _on_check_etcd(self, event):
+        """Check if node is accounted for.
+
+        Check if slurmctld accounted for this node's inventory for the first
+        time, if so, emit slurmctld_started event, so the node can start the
+        daemon.
+        """
+
+        host = self._slurmd.slurmctld_address
+        logger.debug(f"## Connecting to etcd3 in {host}")
+        client = Etcd3Client(host=host, port=2379, api_path="/v3/")
+
+        logger.debug("## Querying etcd3 for node list")
+        v = client.get(key="all_nodes")
+        logger.debug(f"## Got: {v}")
+
+        node_accounted = False
+        if v:
+            hostnames = eval(v[0])
```

Comment:
This is kinda strange. What is contained in the "all_nodes" key? Maybe there's a way to do this that is more transparent?

Reply:
it is a `repr()` of a list of strings. We need to store the list of hostnames in there, but it only accepts a string, not a list. What do you suggest?